codegen/src/test/resources/software/amazon/awssdk/codegen/poet/model/PaginatedOperationWithResultKeyPaginator.java

```diff
@@ -0,0 +1,97 @@
+package software.amazon.awssdk.services.jsonprotocoltests.paginators;
+
+import java.util.Iterator;
+import java.util.function.Function;
+import javax.annotation.Generated;
+import software.amazon.awssdk.core.pagination.NextPageFetcher;
+import software.amazon.awssdk.core.pagination.PaginatedItemsIterable;
+import software.amazon.awssdk.core.pagination.PaginatedResponsesIterator;
+import software.amazon.awssdk.core.pagination.SdkIterable;
+import software.amazon.awssdk.services.jsonprotocoltests.JsonProtocolTestsClient;
+import software.amazon.awssdk.services.jsonprotocoltests.model.PaginatedOperationWithResultKeyRequest;
+import software.amazon.awssdk.services.jsonprotocoltests.model.PaginatedOperationWithResultKeyResponse;
+import software.amazon.awssdk.services.jsonprotocoltests.model.SimpleStruct;
+
+/**
+ * Represents the output for the
```

Comment:
This is some really nice Javadoc! Would it be more discoverable on the operation method itself?

Reply:
Will add docs in client too. Was figuring out some formatting issues